### PR TITLE
Add calendar event on date-cell click

### PR DESCRIPTION
**Pull request by CapceleaCristian**
## Changelog

```changelog
- Add styled component
- Add possibility to save events in LocalStorage 
- Add spin loader on calendar
- Add onClick on a cell, modal pop-up to add new event.
- On DoubleClick delete event
```

## Change Type

- [x] `Added`: new feature - non-breaking change which adds functionality.
- [ ] `Bug fix`: non-breaking change which fixes an issue.
- [x] `Changed`: for changes in existing functionality.
- [ ] `Deprecated`: for soon-to-be removed features.
- [ ] `Documentation` update.
- [ ] `Performance`: for performance and optimization functionality.
- [ ] `Removed`: for now removed features.
- [ ] `Security`: in case of vulnerabilities.

## Required Checks

- [ ] Documentation
- [ ] Unit tests
- [x] Tested on Chrome
- [x] Tested on Firefox
- [x] Tested on IE11
- [ ] npm run eslint
- [ ] npm run prettier

## Screenshots

---

### CODE REVIEW

File: src/features/calendar-events/index.js [hunk 1]
```diff
@@ -1,33 +1,65 @@
...
+      start: new Date('2019, 12, 31'),
+      end: new Date('2020, 01, 01'),
+      title: 'Happy New Year',
+      allDay: true,
+    },
+  ])
+
+  const handleSelect = ({ start, end }) => {
+    if (start.getDate() >= defaultDate.getDate()) {
+      // eslint-disable-next-line no-alert
```
Comment (utopiaeh) [inline src/features/calendar-events/index.js]: Why we use eslint if you disable it

Comment (CapceleaCristian) [reply src/features/calendar-events/index.js]: Unexpected prompt, idk

Comment (utopiaeh) [reply src/features/calendar-events/index.js]: `https://eslint.org/docs/rules/no-alert`

**Review by utopiaeh — APPROVED** [review]

File: src/features/calendar-events/calendar-modal/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { Input, Modal } from 'antd'
+import { bool, func, string } from 'prop-types'
+
+const EventModal = ({ visible, onOk, onCancel, value, onChange }) => {
+  return (
```
Comment (andreiturcanu) [inline src/features/calendar-events/calendar-modal/index.js]: same thing. Remove return

**Review by andreiturcanu — APPROVED** [review]

**Review by denisgluscenco — APPROVED** [review]

**Review by utopiaeh — APPROVED** [review]

**Review by denisgluscenco — APPROVED** [review]

**Review by mirabela-t — APPROVED** [review]

File: src/features/calendar-events/index.js [hunk 3]
```diff
@@ -1,33 +1,103 @@
...
+    if (start.getDate() >= defaultDate.getDate()) {
+      setModalVisibility(true)
+      setCurrentEvent({ start, end })
+    }
+  }
+
+  const handleOk = () => {
+    setModalVisibility(false)
+    if (inputValue) {
+      setCurrentEvent(Object.assign(currentEvent, { title: inputValue }))
```
Comment (enatofan) [inline src/features/calendar-events/index.js]: You should not mutate state variables directly, use either `setCurrentEvent(Object.assign({}, currentEvent, { title: inputValue }))` or to be more consistent with the rest of the code, use the **spread** operator `setCurrentEvent({...currentEvent, title: inputValue})`.

**Review by enatofan — CHANGES_REQUESTED** [review]
Fix the line I commented on :))

**Review by enatofan — APPROVED** [review]

**Pull request merged**